sympy/solvers/ode/riccati.py

```diff
@@ -0,0 +1,465 @@
+        coeffs = list(map(lambda x: x[1], coeffs))
+    else:
+        coeffs = list(map(lambda x: x/den, num.as_dict().values()))
+    return coeffs[::-1] if reverse else coeffs
+
+
+def compute_degree(x, poles, choice, c, d, N):
+    ybar = 0
+    m = S(d[choice[0]][-1])
+
+    # Calculate the first (nested) summation for ybar
+    # as given in Step 9 of the Thesis (Pg 82)
+    for i in range(len(poles)):
+        for j in range(len(c[i][choice[i + 1]])):
+            # If one of the poles is infinity and its coefficient is
+            # not zero, the given solution is invalid as there will be
+            # a c/(x - oo)^j term in ybar for some c and j
+            if poles[i] == oo and c[i][choice[i + 1]][j] != 0:
+                return m, ybar, False
+            ybar += c[i][choice[i + 1]][j]/(x - poles[i])**(j+1)
+        m -= c[i][choice[i + 1]][0]
+
+    # Calculate the second summation for ybar
+    for i in range(N+1):
+        ybar += d[choice[0]][i]*x**i
+    return m, ybar, True
+
+
+def solve_aux_eq(a, x, m, ybar):
+    p = Function('p')
+    # Eq (5.16) in Thesis - Pg 81
+    auxeq = numer((p(x).diff(x, 2) + 2*ybar*p(x).diff(x) + (ybar.diff(x) + ybar**2 - a)*p(x)).together())
+
+    # Assume that the solution is of the type
+    # p(x) = C0 + C1*x + ... + Cm*x**m
+    psyms = get_numbered_constants(auxeq, m)
+    if type(psyms) != tuple:
+        psyms = (psyms, )
+    psol = x**m
+    for i in range(m):
+        psol += psyms[i]*x**i
+    if m != 0:
+        # m is a non-zero integer. Find the constant terms using undetermined coefficients
+        return psol, solve_undetermined_coeffs(auxeq.subs(p(x), psol).doit().expand(), psyms, x), True
+    else:
+        # m == 0 . Check if 1 (x**0) is a solution to the auxiliary equation
+        cf = auxeq.subs(p(x), psol).doit().expand()
+        return S(1), cf, cf == 0
+
+
+def solve_riccati(eq, fx, x, b0, b1, b2):
+    # Step 1 : Convert to Normal Form
+    a = -b0*b2 + b1**2/4 - b1.diff(x)/2 + 3*b2.diff(x)**2/(4*b2**2) + b1*b2.diff(x)/(2*b2) - b2.diff(x, 2)/(2*b2)
+    a_t = a.together()
+    num, den = map(lambda e: Poly(e, x), a_t.as_numer_denom())
+    num, den = num.cancel(den, include=True)
+
+    # Step 2
+    presol = []
+
+    # Step 3 : a(x) is 0
+    if num == 0:
+        presol.append(1/(x + get_numbered_constants(eq, 1)))
+
+    # Step 4 : a(x) is a non-zero constant
+    elif x not in num.free_symbols.union(den.free_symbols):
+        presol.extend([sqrt(a), -sqrt(a)])
+
+    # Step 5 : Find poles and valuation at infinity
+    poles = find_poles(num, den, x)
+    inf_mul = poles.get(oo, 0)
+    poles, muls = list(poles.keys()), list(poles.values())
+    val_inf = val_at_inf(num, den, x)
+
+    if len(poles):
+        # Check necessary conditions (outlined in the module docstring)
+        if val_inf%2 != 0 or not all(map(lambda mul: (mul == 1 or (mul%2 == 0 and mul >= 2)), muls)):
+            raise ValueError("Rational Solution doesn't exist")
```

Comment:
It would be good to put these conditions in a function with a docstring that clearly explains what they mean